minifi-c2/minifi-c2-commons/src/main/java/org/apache/nifi/minifi/c2/model/extension/DefinedType.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.minifi.c2.model.extension;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * A reference to a defined type identified by bundle and fully qualified class type identifiers
+ */
+@ApiModel
+public class DefinedType {
+
+    private String group;
+    private String artifact;
+    private String version;
+    private String type;
```

Comment:
A defined type is effectively a version for a class that's deployable or any class in the software stack?

Reply:
I think in practice it will just be a class that is deployable, i.e., via a flow config.yaml. So basically a processor, controller service, or reporting task.

But it could be more than that. For example, in NiFi, some of the core framework interface implementations are actually built as NARs and deployed by including them in the /lib dir. So givent that the model in the `org.apache.nifi.minifi.c2.model.extension` package is designed to be a prototype for a model that could eventually live in an Extension Registry, these could be used in the NiFi case as well. But for our purposes, just the types that would be used as Flow Components